### DISCUSSION TITLE
How to use CSS variables for media query breakpoints

### DISCUSSION THREAD

**Discussion by Angry-Sparrow** [vercel/next.js#59224] [Help] [answered]
### Summary

I defined a breakpoint value to adapt through media queries.But css variables didn't work in @media query.
`
.fullscreenBanner {
  width: 100%;
  height: 100vh;
  position: relative;
  background: #d8d8d8;

  @media (max-width: var(--sm-screen-max-width)) {
    width: 80%;
  }
}
`

However, it works in normal css description.
`{
 .myClass {
  width: var(--sm-screen-max-width);
}
}`

`--sm-screen-max-width`  has been declared in global.css.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Yeah, I think the first spec of CSS vars can't do that:

- https://bholmes.dev/blog/alternative-to-css-variable-media-queries/
- https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties

> Note: Variables do not work inside media queries and container queries. You can use the [var()](https://developer.mozilla.org/en-US/docs/Web/CSS/var) function in any part of a value in any property on an element. You cannot use [var()](https://developer.mozilla.org/en-US/docs/Web/CSS/var) for property names, selectors, or anything aside from property values, which means you can't use it in a media query or container query.

There's however, CSS pre-processor tooling that can help you achieve this, and there's an upcoming CSS feature called, `@custom-media`, which wants to do this very thing:

- https://drafts.csswg.org/mediaqueries-5/#custom-mq

**Reply by Angry-Sparrow** [reply]
Thank you very much, icyJoseph.
